### PR TITLE
Post describer request

### PR DESCRIPTION
**Pull request by calummackervoy**
adds `describeScene` method and incorporates `describeThing` into this

### CODE REVIEW

File: lib/PerceptionManager.tsx [hunk 1]
```diff
@@ -59,47 +64,82 @@ export const perceptionManager: IPerceptionManager = (() => {
...
+        
+        for(let thing of things) {
+            const uri = asUrl(thing as ThingPersisted);
+            const imageUrl = getUrl(thing, MUD.primaryImageContent);
+
+            if(imageUrl && !recentUris.includes(uri)) {
+                messages.push(getITerminalMessage(<img src={imageUrl}></img>));
+            }
+    
+            const msg = (getThingName(thing) + ". " || "") + (getStringNoLocale(thing, MUD.primaryTextContent) || "");
```
Comment (MattTennison) [inline lib/PerceptionManager.tsx]: What about:
```
const isPresent = value => Boolean(value)
const msg = [getThingName(thing), getStringNoLocale(thing, MUD.primaryTextContent)].filter(isPresent).join(". ")
```

That :point_up: wouldn't behave exactly the same, if getStringNoLocale was falsy but getThingName was:
Current Output: `thingname. `
Output with this change: `thingname`

But I think that's what you want?

File: lib/PerceptionManager.tsx [hunk 2]
```diff
@@ -59,47 +64,82 @@ export const perceptionManager: IPerceptionManager = (() => {
...
+     * builds a list of ITerminalMessage objects by getting the primary content data from parameterised things
+     */
+    const getPrimaryContent = (things: Thing[]) : ITerminalMessage[] => {
+        let messages: ITerminalMessage[] = [];
+        
+        for(let thing of things) {
+            const uri = asUrl(thing as ThingPersisted);
+            const imageUrl = getUrl(thing, MUD.primaryImageContent);
+
+            if(imageUrl && !recentUris.includes(uri)) {
```
Comment (MattTennison) [inline lib/PerceptionManager.tsx:87]: This conditional - do we actually want msg below to be added if this is falsy?

**Review by MattTennison — APPROVED** [review]
Looks good. If the answer to my question re: conditional is "yes we do want it" then feel free to merge :smile:

File: lib/PerceptionManager.tsx (hunk 2 above)
Comment (calummackervoy) [reply lib/PerceptionManager.tsx:87]: I think so most of the time, e.g. if I visited the bar a minute ago and then I go back in, I still want it to be described

It'd be nice to have it as a parameter to this function I suppose, and not even "on/off" but maybe something like `ContentLevelEnum` ... but I think that's a future development

File: lib/PerceptionManager.tsx (hunk 1 above)
Comment (calummackervoy) [reply lib/PerceptionManager.tsx]: Nice style :)

**Pull request merged**